File: book/petrol-pump/java/chapter4/section4/LifeCyclePump.java
package chapter4.section4;

import pump.*;
import nz.sodium.*;
import java.util.Optional;

public class LifeCyclePump implements Pump {
    public Outputs create(Inputs inputs) {
        LifeCycle lc = new LifeCycle(inputs.sNozzle1,
                                     inputs.sNozzle2,
                                     inputs.sNozzle3);
        return new Outputs()
            .setDelivery(lc.fillActive.map(
                of ->
                    of.equals(Optional.of(Fuel.ONE))   ? Delivery.FAST1 :
                    of.equals(Optional.of(Fuel.TWO))   ? Delivery.FAST2 :
                    of.equals(Optional.of(Fuel.THREE)) ? Delivery.FAST3 : 
                                                         Delivery.OFF))
            .setSaleQuantityLCD(lc.fillActive.map(
                of ->
                    of.equals(Optional.of(Fuel.ONE))   ? "1" :
                    of.equals(Optional.of(Fuel.TWO))   ? "2" :
                    of.equals(Optional.of(Fuel.THREE)) ? "3" : ""));
    }
}

File: book/petrol-pump/c#/PetrolPump/Chapter4/Section4/LifeCyclePump.cs
using Sodium.Frp;
using Sodium.Functional;

namespace PetrolPump.Chapter4.Section4
{
    public class LifeCyclePump : IPump
    {
        public Outputs Create(Inputs inputs)
        {
            LifeCycle lc = new LifeCycle(inputs.SNozzle1, inputs.SNozzle2, inputs.SNozzle3);
            return new Outputs()
                .SetDelivery(lc.FillActive.Map(m =>
                    m.Equals(Maybe.Some(Fuel.One)) ? Delivery.Fast1 :
                        m.Equals(Maybe.Some(Fuel.Two)) ? Delivery.Fast2 :
                            m.Equals(Maybe.Some(Fuel.Three)) ? Delivery.Fast3 :
                                Delivery.Off))
                .SetSaleQuantityLcd(lc.FillActive.Map(m =>
                    m.Equals(Maybe.Some(Fuel.One)) ? "1" :
                        m.Equals(Maybe.Some(Fuel.Two)) ? "2" :
                            m.Equals(Maybe.Some(Fuel.Three)) ? "3" :
                                string.Empty));
        }
    }
}